File: FunnyAlgorithms-master/prime_checker/prime_checker.py
# python script to check if a given number is prime
def isPrime(n):
	#first check if number is 2 or 3
	if n <= 3:
		return n > 1
	#check if divisible by 2 or 3
	elif n%2 == 0 or n%3 == 0:
		return False

	#since all numbers divisble by 2 and 3 are covered, the next prime dividor is 5
	#so this only checks for numbers bigger than 25
	c = 5
	while (c*c <= n):
		if n%c == 0 or n%(c+2)==0:
			return False
		c += 6

	#if all checks pass then it is a prime
	return True


#test statements
result = isPrime(5)
print(str(result) + ' should be True')
result = isPrime(10273)
print(str(result) + ' should be True')
result = isPrime(50)
print(str(result) + ' should be False')

File: prime_checker/prime_checker.js
const isPrime = function(x) {

if(x%2===0 && x!=2 || x%3===0 && x!=3 || x%5===0 && x!=5|| x%7===0 && x!=7 || x%11===0 && x!=11){
  return false;
}
else return true;
}


//Test
console.log(isPrime(5));
console.log(isPrime(10273));
console.log(isPrime(50));
